src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,340 @@
+                log.error(errorMessage, e);
+                dataStoreToArmHelper.recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount);
+                errorMessage = errorMessage + " - " + ExceptionUtils.getStackTrace(e);
+                updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessage);
+            }
+        }
+
+        if (!writeManifestAndCopyToArm(userAccount, batchItems, archiveRecordsFile)) {
+            return;
+        }
+
+        for (var batchItem : batchItems.getSuccessful()) {
+            dataStoreToArmHelper.updateExternalObjectDirectoryStatus(batchItem.getArmEod(), EodHelper.armDropZoneStatus(), userAccount);
+            logApi.armPushSuccessful(batchItem.getArmEod().getId());
+        }
+        
+    }
+
+    private boolean writeManifestAndCopyToArm(UserAccountEntity userAccount, ArmBatchItems batchItems, File archiveRecordsFile) {
+        try {
+            if (!batchItems.getSuccessful().isEmpty()) {
+                dataStoreToArmHelper.writeManifestFile(batchItems, archiveRecordsFile);
+                updateObjectStateRecordManifestSuccessOrFailure(batchItems, archiveRecordsFile);
+                copyMetadataToArm(archiveRecordsFile);
+            } else {
+                log.warn("No EODs were able to be processed, skipping manifest file creation");
+            }
+        } catch (Exception e) {
+            String errorMessage = String.format("Error during generation of DETS batch manifest file %s", archiveRecordsFile.getName());
+            log.error(errorMessage, e);
+            batchItems.getSuccessful().forEach(batchItem -> dataStoreToArmHelper.recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount));
+            final String errorMessageWithStackTrace = errorMessage + " - " + ExceptionUtils.getStackTrace(e);
+            batchItems.getFailed().forEach(batchItem -> updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessageWithStackTrace));
+            return false;
+        }
+        return true;
+    }
+
+    private String getManifestFilePrefix() {
+        return detsToArmProcessorConfiguration.getManifestFilePrefix();
+    }
+
+    private void updateObjectStateRecordManifestSuccessOrFailure(ArmBatchItems batchItems, File archiveRecordsFile) {
+        for (var batchItem : batchItems.getSuccessful()) {
+            ObjectStateRecordEntity objectStateRecord = getObjectStateRecordEntity(batchItem.getArmEod());
+            if (nonNull(objectStateRecord)) {
+                objectStateRecord.setFlagFileMfstCreated(true);
+                objectStateRecord.setDateFileMfstCreated(currentTimeHelper.currentOffsetDateTime());
+                objectStateRecord.setIdManifestFile(archiveRecordsFile.getName());
+                objectStateRecord.setFlagMfstTransfToArml(true);
+                objectStateRecord.setDateMfstTransfToArml(currentTimeHelper.currentOffsetDateTime());
```

Comment:
the ticket says that these 2 fields should be set only after the manifest file has been pushed to ARM, which hasn't happened at this stage yet if I'm not wrong. 
Should these be moved inside the next step instead?:
```
copyMetadataToArm(archiveRecordsFile);
```